packages/text-input/src/TextInput.tsx

```diff
@@ -286,10 +286,10 @@ const TextInput = React.forwardRef(
         {description && <p className={descriptionStyle}>{description}</p>}
         <div className={inputContainerStyle}>
           <input
+            type="text"
```

Comment:
To note: the type will error if it's not compliant with expected type values on an HTML input element, as the props are extending those of an HTMLInputElement